src/backends/backend.js

```diff
@@ -133,6 +133,7 @@ class Backend {
     const extension = selectFolderEntryExtension(collection);
     const collectionFilter = collection.get('filter');
     return listMethod.call(this.implementation, collection, extension)
+      .then(loadedEntries => loadedEntries.filter(loadedEntry => !loadedEntry.error))
```

Comment:
Filtering out entries that failed to load seems like an implementation detail of the backend to me, and signalling by resolving to an object with an `error` property is pretty confusing, especially since it requires coordinated keys in two different files. I would suggest one of two refactors:

- This `.then(x => x.filter(...))` line could be moved into `src/backends/github/implementation.js`, right after the [`return Promise.all(...)` line](https://github.com/netlify/netlify-cms/pull/1093/files#diff-5516348e32db81ebc1300ac5ce8d047bL82). (This is the refactor that seems best to me.)
- You could do all the filtering here by dropping the `Promise.all` in `fetchFiles` and returning an array of `Promise`s instead, then map `.catch` over them here, instead of in `src/backends/github/implementation.js`. (I don't like this refactoring as much - catching these errors seems like a detail that should be handled by the backends.)

Reply:
Agreed, I'll move it.